tests/align/align_create_tensor_torch.py

```diff
@@ -0,0 +1,646 @@
+        label, output = create_tensors_for_scalar_multiply_torch()
+    elif operator_name == 'scalar_sub':
+        label, output = create_tensors_for_scalar_sub_torch()
+    elif operator_name == 'scalar_truediv':
+        label, output = create_tensors_for_scalar_truediv_torch()
+    elif operator_name == 'sigmoid':
+        label, output = create_tensors_for_sigmoid_torch()
+    elif operator_name == 'sin':
+        label, output = create_tensors_for_sin_torch()
+    elif operator_name == 'subtract':
+        label, output = create_tensors_for_subtract_torch()
+    elif operator_name == 'tanh':
+        label, output = create_tensors_for_tanh_torch()
+    elif operator_name == 'transpose':
+        label, output = create_tensors_for_transpose_torch()
+    elif operator_name == 'view_embedding':
+        label, output, weight = create_tensors_for_scalar_view_embedding_torch(param_dir=OUT_DIR)
+    else:
+        raise ValueError('Not Include such Operator in Aligment Test ', operator_name)
+
+    output.retain_grad()
+    loss_fn = torch.nn.MSELoss(reduction="mean")
+    loss = loss_fn(output, label)
+    loss.backward()
+
+    # save tensors to file
+    torch.save(output.cpu(), os.path.join(OUT_DIR, "torch_out.pt"))
+    torch.save(output.grad.cpu(), os.path.join(OUT_DIR, "torch_out_grad.pt"))
+
+    if (operator_name in param_weight_op):
+        torch.save(weight.grad.cpu(), os.path.join(
+            OUT_DIR, "torch_weight_grad.pt"))
+    if (operator_name in param_bias_op):
+        torch.save(bias.grad.cpu(), os.path.join(
+            OUT_DIR, "torch_bias_grad.pt"))
+
+# run tests for all operators
+
+
+def create_tensors_for_add_torch():
+    inp1: torch.Tensor = gen_tensor(
+        (BATCH_SIZE, SEQ_LENGTH, INPUT_SIZE),
+        dtype="float32"
+    ).to(DEVICE)
+    inp2: torch.Tensor = gen_tensor(
+        (BATCH_SIZE, SEQ_LENGTH, INPUT_SIZE),
+        dtype="float32"
+    ).to(DEVICE)
+    label: torch.Tensor = gen_tensor(
+        (BATCH_SIZE, SEQ_LENGTH, INPUT_SIZE),
+        dtype="float32"
+    ).to(DEVICE)
+    output = torch.add(
+        input=inp1,
+        other=inp2
+    ).to(DEVICE)
+    output.requires_grad = True
+    return label, output
+
+
+def create_tensors_for_concat_torch():
+    inp1: torch.Tensor = gen_tensor(
+        (BATCH_SIZE, SEQ_LENGTH, INPUT_SIZE),
+        dtype="float32"
+    ).to(DEVICE)
+    inp2: torch.Tensor = gen_tensor(
+        (BATCH_SIZE, SEQ_LENGTH, INPUT_SIZE),
+        dtype="float32"
+    ).to(DEVICE)
+    inp3: torch.Tensor = gen_tensor(
+        (BATCH_SIZE, SEQ_LENGTH, INPUT_SIZE),
+        dtype="float32"
+    ).to(DEVICE)
+    label: torch.Tensor = gen_tensor(
+        (BATCH_SIZE, SEQ_LENGTH * 3, INPUT_SIZE),
+        dtype="float32"
+    ).to(DEVICE)
+    output = torch.cat(
+        tensors=(inp1, inp2, inp3),
+        dim=1
+    ).to(DEVICE)
+    output.requires_grad = True
+
+    return label, output
+
+
+def create_tensors_for_conv2d_torch(param_dir):
```

Comment:
Why is this not called in the big switch-case function above?

Reply:
yes, some of them are identical, some are not. thanks for reminding. solved by created function in utils